packages/next/src/build/index.ts

```diff
@@ -2049,16 +2049,19 @@ export default async function build(
                             isInterceptionRouteAppPath(page)
                           if (appConfig.revalidate !== 0) {
                             const isDynamic = isDynamicRoute(page)
-                            const hasGenerateStaticParams =
-                              !!workerResult.prerenderRoutes?.length
+                            const generateStaticParamsIsDefined =
+                              !!workerResult.prerenderRoutes
+                            const generateStaticParamsHasItems =
+                              generateStaticParamsIsDefined &&
+                              workerResult.prerenderRoutes?.length
 
                             if (
                               config.output === 'export' &&
                               isDynamic &&
-                              !hasGenerateStaticParams
+                              !generateStaticParamsIsDefined
                             ) {
```

Comment:
A new value is created for this check alone in case the current behavior is necessary for other parts of the build process.